src/auth/UserRequest.cc

```diff
@@ -132,6 +132,18 @@ Auth::UserRequest::denyMessage(char const * const default_message) const
     return getDenyMessage();
 }
 
+bool Auth::UserRequest::authenticated() const
+{
+    if (user() != nullptr && user()->credentials() == Auth::Ok)
```

Comment:
Please do not call user() twice, especially because it is virtual.

Reply:
done